### PR TITLE
Lifts BFI and BFXIL

### PR DESCRIPTION
**Pull request by toshipiazza**

### CODE REVIEW

File: il.cpp [hunk 1]
```diff
@@ -757,6 +757,30 @@ bool GetLowLevelILForInstruction(Architecture* arch, uint64_t addr, LowLevelILFu
...
+				il.And(REGSZ(operand1),
+					il.Const(REGSZ(operand1), ~(((1LL << IMM(operand4)) - 1) << IMM(operand3))),
+					ILREG(operand1)),
+				il.ZeroExtend(REGSZ(operand1), il.ShiftLeft(REGSZ(operand2),
+					il.And(REGSZ(operand2), ILREG(operand2), il.Const(REGSZ(operand2), (1LL << IMM(operand4)) - 1)),
+						il.Const(1, IMM(operand3)))))));
+		break;
+	case ARM64_BFXIL:
+	{
+		// prevent shifting by 64 bits which is UB
```
Comment (toshipiazza) [inline il.cpp:772]: BFI doesn't have this edge case. The only instructions for which width = 64 are:
```
bfxil x2, x3, 0, 0x40
bfi   x2, x3, 0, 0x40
```

Surprisingly, both instructions have the same encoding, and our disassembler breaks the tie with BFXIL.

Should I add this as a comment?

**Comment by toshipiazza** [comment]
Since binja lifts BFC as BFI (why?), I believe this also works for BFI

```
bfi     x2, xzr, #0x10, #0x10
bfi     w2, wzr, #0x10, #0x10
```

The IL seems correct:

```
x2 = (-0xffff0001 & x2) | zx.q(0 << 0x10)
w2 = (-0xffff0001 & w2) | zx.d(0 << 0x10)
```

**Comment by toshipiazza** [comment]
XREF #2

**Comment by lwerdna** [comment]
@toshipiazza You found that BFC is not being correctly disassembled. My incorrect answer in slack last night have caused you some unnecessary work :(. Let me fix _disassembly_ of BFC first then we'll check your lifting. Thanks.

**Comment by lwerdna** [comment]
@toshipiazza Thank you for the pull. Please include tests next time (in arm64test.py). I adjusted the BFI lift a bit, removing the zero extend. I think it is ok but its possibly I overlooked something.

**Comment by lwerdna** [comment]
Oh I overlooked something alright! 😬 Give me a bit to fix this.

**Comment by toshipiazza** [comment]
Will do next time 😃

**Pull request merged**